lib/ansible/modules/cloud/amazon/sqs_queue.py

```diff
@@ -258,58 +363,95 @@ def set_queue_attribute(queue, attribute, value, check_mode=False):
+
+
+def modify_tags(queue_url, connection, module, result, tags, purge_tags):
+    current_tags = None
+    changed = False
+
+    # Get current tags if they exist
+    try:
+        current_tags = connection.list_queue_tags(QueueUrl=queue_url)['Tags']
+    except:
+        tags_need_modify = tags
+
+    # Modify current tags
+    if current_tags:
+        tags_need_modify, tags_to_delete = compare_aws_tags(current_tags, tags, purge_tags)
+
+        if tags_to_delete:
+            try:
+                connection.untag_queue(QueueUrl=queue_url, TagKeys=tags_to_delete)
+                changed = True
+            except botocore.exceptions.ClientError as e:
+                result['msg'] = 'Failed to remove tags: ' + traceback.format_exc()
+                module.fail_json(**result)
+
+    # Add/update tags
+    if tags_need_modify:
+        try:
+            connection.tag_queue(QueueUrl=queue_url, Tags=tags_need_modify)
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            result['msg'] = 'Failed to add tags: ' + traceback.format_exc()
+            module.fail_json(**result)
+
+    return changed
 
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
         state=dict(default='present', choices=['present', 'absent']),
         name=dict(required=True, type='str'),
+        queue_type=dict(default='standard', choices=['standard', 'fifo'], required=False),
+        fifo_content_based_deduplication=dict(default=False, type='bool', required=False),
         default_visibility_timeout=dict(type='int'),
         message_retention_period=dict(type='int'),
         maximum_message_size=dict(type='int'),
         delivery_delay=dict(type='int'),
         receive_message_wait_time=dict(type='int'),
         policy=dict(type='dict', required=False),
         redrive_policy=dict(type='dict', required=False),
+        tags=dict(default={}, type='dict', required=False),
+        purge_tags=dict(default=True, type='bool', required=False)
     ))
 
-    module = AnsibleModule(
+    module = AnsibleAWSModule(
         argument_spec=argument_spec,
         supports_check_mode=True)
 
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
-
-    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
-    if not region:
-        module.fail_json(msg='region must be specified')
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 and botocore are required for this module')
 
     try:
-        connection = connect_to_aws(boto.sqs, region, **aws_connect_params)
-
-    except (NoAuthHandlerFound, AnsibleAWSError) as e:
-        module.fail_json(msg=str(e))
+        region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+        connection = boto3_conn(module, conn_type='client', resource='sqs', region=region, endpoint=ec2_url, **aws_connect_params)
+    except botocore.exceptions.NoRegionError:
```

Comment:
boto3_conn now handles region problems, no need to do it in the module

Reply:
Is there any documentation I can look up about this?  Struggled to find any :)